### PR TITLE
[BD-46] feat: table of content for non-components

### PR DESCRIPTION
**Pull request by monteri**
## Description

Table of content for other than components pages. It is gathered by traversing heading elements on the page.

### Deploy Preview

https://deploy-preview-2013--paragon-openedx.netlify.app/

## Merge Checklist

* [x] If your update includes visual changes, have they been reviewed by a designer? Send them a link to the Netlify deploy preview, if applicable.
* [x] Does your change adhere to the documented [style conventions](https://github.com/openedx/paragon/blob/master/docs/decisions/0012-css-styling-conventions)?
* [x] Do any prop types have missing descriptions in the Props API tables in the documentation site (check deploy preview)?
* [x] Were your changes tested using all available themes (see theme switcher in the header of the deploy preview, under the "Settings" icon)?
* [x] Were your changes tested in the `example` app?
* [x] Is there adequate test coverage for your changes?
* [x] Consider whether this change needs to reviewed/QA'ed for accessibility (a11y). If so, please add `wittjeff` and `adamstankiewicz` as reviewers on this PR.

## Post-merge Checklist

* [x] Verify your changes were released to [NPM](https://www.npmjs.com/package/@edx/paragon) at the expected version.
* [x] If you'd like, [share](https://github.com/openedx/paragon/discussions/new?category=show-and-tell) your contribution in [#show-and-tell](https://github.com/openedx/paragon/discussions/categories/show-and-tell).
* [x] 🎉 🙌 Celebrate! Thanks for your contribution.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @monteri!


When this pull request is ready, tag your edX technical lead.

**Review by PKulkoRaccoonGang — APPROVED** [review]

**Review by PKulkoRaccoonGang — APPROVED** [review]

File: example/src/MyComponent.jsx [hunk 1]
```diff
@@ -1,8 +1,8 @@
 import React, { useState } from 'react';
-import { Button, Form, Icon, Bubble } from '@edx/paragon'; // eslint-disable-line
+import { Button, Form, Icon, Bubble, Skeleton } from '@edx/paragon'; // eslint-disable-line
 import { FavoriteBorder } from '@edx/paragon/icons'; // eslint-disable-line
 
-const MyComponent = () => {
+function MyComponent() {
```
Comment (adamstankiewicz) [inline example/src/MyComponent.jsx:5]: Looks like this PR is now failing on a lint error here.

File: www/package.json [hunk 2]
```diff
@@ -38,7 +38,8 @@
     "rehype-slug": "^4.0.1",
     "sass": "^1.53.0",
     "sass-loader": "12.6.0",
-    "uuid": "^9.0.0"
+    "uuid": "^9.0.0",
```
Comment (adamstankiewicz) [inline www/package.json:41]: [curious] The `www/package-lock.json` file appears to be newly generated in this PR. Given the repo is now using NPM workspaces, I don't believe an individual package like `www` should have a `package-lock.json` in it, given its packages get listed/hoisted into the root `package-lock.json` file.

Should be able to delete the generated `www/package-lock.json` file and run `npm i` in the root of the repo?

**Review by adamstankiewicz — COMMENTED** [review]
The ToC and the code all look good. Just failing on a few linting errors and the erroneous creation of the `www/package-lock.json` file.

**Comment by adamstankiewicz** [comment]
@monteri It looks like the container widths and positioning might have regressed in this work? 

### Current state on `master`

image

### Deploy preview 

image

**Review by adamstankiewicz — APPROVED** [review]
Looks great! Thanks!

**Comment by openedx-webhooks** [comment]
@monteri 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/paragon&entry.867055334=https://github.com/openedx/paragon/pull/2013&entry.1484655318=https://github.com/monteri&entry.752974735=2023-02-13+14:33&entry.1917517419=2023-08-18+11:44&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**